.github/workflows/cloud-data-sources-code-coverage.yml

```diff
@@ -0,0 +1,81 @@
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17.8'
+      - name: generate go for github.com/grafana/grafana
+        run: make gen-go
+
+      - name: Get Go coverage for PR
+        run: |
+          go test -v -coverpkg=./... -coverprofile=profile.cov ./pkg/tsdb/azuremonitor/... ./pkg/tsdb/cloudmonitoring/... ./pkg/tsdb/cloudwatch/...
+          echo "go-pr-cov=$(go tool cover -func profile.cov | grep total: | awk '{print substr($NF,1,length($NF)-1)}')" >> $GITHUB_ENV
+      - name: Get TypeScript coverage for PR
+        run: |
+          echo "ts-pr-cov=$(yarn test:coverage:cloud-data-sources | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      - name: yarn install for caller's main
+        run: yarn install
+      - name: generate go for github.com/grafana/grafana
+        run: make gen-go
+
+      - name: Get Go coverage on main
+        run: |
+          go test -v -coverpkg=./... -coverprofile=profile.cov ./pkg/tsdb/azuremonitor/... ./pkg/tsdb/cloudmonitoring/... ./pkg/tsdb/cloudwatch/...
+          echo "go-main-cov=$(go tool cover -func profile.cov | grep total: | awk '{print substr($NF,1,length($NF)-1)}')" >> $GITHUB_ENV
+      - name: Get TypeScript coverage on main
+        run: |
+          echo "ts-main-cov=$(yarn test:coverage:cloud-data-sources | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
+      - name: Calculate coverage difference
+        run: |
+          echo "go-cov-difference=$(bc <<< '${{ env.go-pr-cov }} - ${{ env.go-main-cov }}')" >> $GITHUB_ENV
+          echo "ts-cov-difference=$(bc <<< '${{ env.ts-pr-cov }} - ${{ env.ts-main-cov }}')" >> $GITHUB_ENV
+      - name: Find previous comment, if any
+        uses: peter-evans/find-comment@v2
+        id: find-comment
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: |
+            Code coverage report for PR #${{ github.event.pull_request.number }}
+      - name: Create or update comment
+        uses: peter-evans/create-or-update-comment@v2
+        # Check if the event is not triggered by a fork
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        with:
+          comment-id: ${{ steps.find-comment.outputs.comment-id }}
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            Code coverage on cloud data sources for PR #${{ github.event.pull_request.number }}
+            |      |          Go                         | TypeScript |
+            |------| ----------------------------------- | ---------------- |
+            | main | ${{ env.go-main-cov }}%             | ${{ env.ts-main-cov }}% |
+            | PR   | ${{ env.go-pr-cov }}%               | ${{ env.ts-pr-cov }}%   |
+            | difference | ${{ env.go-cov-difference }}% | ${{ env.ts-cov-difference }}% |
```

Comment:
Feels weird mixing all 3 datasources together but I understand where you are coming from 😉

Reply:
Something that came up in an earlier discussion with @andresmgot was a situation like a non-cloud-data-source test could exist which called a function in a cloud-data-source file. And because it was "external," its coverage would not count towards the plugin report's coverage.

What that highlights is that this isn't going to be a perfect representation of code coverage. And splitting by plugin for example may give the false impression that this is _precise_. 

I would propose starting very simply and roughly for now and see what enhancements we would like after using it in practice. 

At some point if we go far enough for accuracy and precision, we will simply re-invent the full jest coverage report.